### ISSUE TITLE
Vehicle Limits cheat and Reverse Freefall Coaster

### ISSUE THREAD

**Issue by Batchman17** [OpenRCT2/OpenRCT2#20207]
### Operating System

Windows 10 64bit (I think)

### OpenRCT2 build

v0.4.5-6 (Just a few days old)

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This is a development issue

### Describe the issue

Was asked to report this here ... sorry if I screw it up. Used to be the Vehicle Limits cheat would allow me to do two cars to the train on the Reverse Freefall coaster, allowing the doubling of guests riding. Oddly, recently it will allow me to choose multiple trains (absurd because they would crash into one another), but it will not allow more than a single car to the train, just like without the cheat.

Passing on word.

### Steps to reproduce

Build a Reverse Freefall coaster, and use the cheat menu to try and make a 2 car train. Doesn't seem to allow it.

**Comment by spacek531** [comment]
Can confirm on OpenRCT2, v0.4.4-183-g9ca4e48 (9ca4e48 on develop) provided by GitHub

**Comment by cheweytoo** [comment]
Are you sure this ever worked?

I went back all the way to release 0.2.4 (the oldest Linux build available on openrct2.org), checking several 0.2.x, 0.3.x, 0.4.x releases inbetween, up to current develop: "Disable vehicle limits" didn't enable more than one car per train for the reverse freefall coaster in any of them.

There _are_ some unexpected things going on though, at least on the UI level:
After enabling the cheat, the fields for "number of cars per train" and "number of trains" look like this:
![car_per_train_train](https://github.com/OpenRCT2/OpenRCT2/assets/29800630/a71c221f-4351-464d-a16e-d86b3382ff2e)

Note the "1 car" where it should say "1 train".

Clicking on the "1 car per train" field's `+` _once_ leads to:

![car_per_train_train_plus](https://github.com/OpenRCT2/OpenRCT2/assets/29800630/1937d082-e58a-4b34-819f-fe970f4dfa7c)

So a click in the right field modifies the value in the left one. Not quite the expected result. _Subsequent_ clicks on `+` or `-` in the right "cars per train" field do nothing, in the supposeldy-number-of-trains field on the left they change the number of trains.

And yeah, one can set more than one train, leading to a guaranteed train crash.


The tooltips on those inputs say "Max 1 car" and "Max 1 car per train". I'm not sure how relevant the tooltips are though, I just saw "Max 1 car per train" on a bobsleigh coaster and could set the number of cars per train higher than that without any problems. That might be a separate bug.

**Comment by Batchman17** [comment]
First off, regardless of whether it was different somewhere in the past or not, my understanding of the cheat should allow this.

Secondly, I get interested in (and caught up in) games in waves. So I'll spend a few weeks (or occasionally months) playing something, then I will leave it alone for a couple of years, before returning to it. So I can't say with 100% certainty that my memory is correct about having done this in the past, but as I mentioned in the original post, it was 99%. I can very clearly remember setting up the 2 cars, just because of how limited the throughput was on this coaster type, with only the single car. But yes, it is possible, even if I think it is unlikely, that I am misremembering. But if not, it was probably two or three years ago that I remember doing this. Possibly even longer.

**Comment by cheweytoo** [comment]
> my understanding of the cheat mean it should allow this.

I completely agree. It's just that I was prepared to go hunting for the specific commit that broke it in the moderately recent past, and came up empty.


> But if not, it was probably two or three years ago that I remember doing this. Possibly even longer.

The oldest build I could test was version 0.2.4 from October 2019. Going further back would be quite a bit of work.

> something seems to be a little bit hinkey here.

Oh, clearly.

Do you maybe happen to have old savegame files with trains like that? It might very well be just the editing that's broken.

**Comment by cheweytoo** [comment]
What _does_ work is using trains from different coasters via the "Show vehicles from other track types" cheat.

Meet the 60 seater reverse freefall:


https://github.com/OpenRCT2/OpenRCT2/assets/29800630/02a8c94e-8270-446e-83de-35a4e004388f

**Comment by Batchman17** [comment]
That would work, and I suppose it would be a bit hypocritical to say "That'a cheating!' LOL

I like the look of the original car, I just want 2 (or possibly 3) together to allow for more passengers.

And possibly I could accomplish the same thing with a single car, just using the ride vehicle editor, but still ...

**Comment by mrmbernardi** [comment]
The cause of this is that for some reason `carsPerFlatRide` is set to 1 on the reverse freefall coaster where it has to be 255 on coasters where you want to be able to edit the amount of cars in the train.
